src/resources/Account.js

```diff
@@ -25,144 +25,110 @@ export default class Account extends Resource {
     })
   }
 
-  login (email, password = '') {
-    return new Promise(async (resolve, reject) => {
-      try {
-        const salt = await this.retrieveSalt(email) || ''
-        const hash = await bcrypt.hash(password, salt)
-        const result = await this.validateCredentials(email, hash)
+  async login (email, password = '') {
```

Comment:
`try/catch` is better here. I plan to normalize all errors eventually, but this pattern should be observed.

For now you can:

```js
try {
  const user = await getUser()
} catch (err) {
  throw err
}